WordPress/Classes/ViewRelated/Media/MediaLibraryViewController.swift

```diff
@@ -42,16 +44,80 @@ class MediaLibraryViewController: UIViewController {
 
         title = NSLocalizedString("Media", comment: "Title for Media Library section of the app.")
 
+        updateNavigationItem()
+
         addMediaPickerAsChildViewController()
     }
 
+    private func updateNavigationItem() {
+        if isEditing {
+            navigationItem.setLeftBarButton(UIBarButtonItem(barButtonSystemItem: .cancel, target: self, action: #selector(editTapped)), animated: true)
+            navigationItem.setRightBarButton(UIBarButtonItem(image: Gridicon.iconOfType(.trash), style: .plain, target: self, action: #selector(trashTapped)), animated: true)
+            navigationItem.rightBarButtonItem?.isEnabled = false
+        } else {
+            navigationItem.setLeftBarButton(nil, animated: true)
+            navigationItem.setRightBarButton(UIBarButtonItem(barButtonSystemItem: .edit, target: self, action: #selector(editTapped)), animated: true)
+        }
+    }
+
     private func addMediaPickerAsChildViewController() {
         pickerViewController.willMove(toParentViewController: self)
         pickerViewController.view.bounds = view.bounds
         view.addSubview(pickerViewController.view)
         addChildViewController(pickerViewController)
         pickerViewController.didMove(toParentViewController: self)
     }
+
+    @objc private func editTapped() {
+        isEditing = !isEditing
+
+        pickerViewController.allowMultipleSelection = isEditing
+
+        pickerViewController.clearSelectedAssets(true)
+    }
+
+    @objc private func trashTapped() {
+        let alertController = UIAlertController(title: nil,
+                                                message: NSLocalizedString("Are you sure you want to permanently delete these items?", comment: "Message prompting the user to confirm that they want to permanently delete a group of media items."), preferredStyle: .alert)
```

Comment:
Looking at Calypso, we should also provide a singular form of this text when only selecting one item and pressing delete. `Are you sure you want to permanently delete this item?`